File: Assignment__22/Question4.c
#include<stdio.h>
#include<stdlib.h>

int CountEleven(int Arr[], int iSize)
{
    int iCnt = 0;

    int iElevenCount = 0;

    for(iCnt = 0; iCnt < iSize; iCnt++)    
    {
        if(Arr[iCnt] == 11)
        {
            iElevenCount = iElevenCount + 1;
        }
    }

    return iElevenCount;
}

int main()
{
    int iLength = 0;
    int iCnt = 0;
    int *ptr = NULL;
    int iResult = 0;

    printf("Enter the number of elements : ");
    scanf("%d", &iLength);

    ptr = (int *)malloc(iLength * sizeof(int));
    
    if(ptr == NULL)      
     {
        printf("Unable to allocate the memoey\n");
        return -1;
    }

    printf("Enter the elements : \n");

    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        scanf("%d",&ptr[iCnt]);
    }

    iResult = CountEleven(ptr,iLength);
    printf("The frequency of number 11 in the given array is : %d", iResult);

    free(ptr);

    return 0;
}

File: Assignment__22/Question4.java
//Accept N numbers from user and return frequency of 11 form it. 

import java.util.Scanner;

class ArrayX
{
    public int Arr[];
    public int iSize;

    public ArrayX(int iNo)
    {
        iSize = iNo;

        Arr = new int [iSize];    //Need to remember
    }

    public void Accept()
    {
        Scanner sobj1 = new Scanner(System.in);

        int iCnt = 0;

        System.out.println("Enter the numbers : ");

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            Arr[iCnt] = sobj1.nextInt();
        }
    }

    public void Display()
    {
        System.out.println("The elements of the array are : ");

        int iCnt = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            System.out.println(Arr[iCnt]);
        }
    }

    public int ElevenCount()
    {
        int iCnt = 0;
        int iCount = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if(Arr[iCnt] == 11)
            {
                iCount++;
            }   
        }
        return iCount;
    }
}

class Question4
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter the number of elements : ");
        int iValue = sobj.nextInt();

        int iResult = 0;

        ArrayX aobj = new ArrayX(iValue);
        aobj.Accept();
        aobj.Display();

        iResult = aobj.ElevenCount();

        System.out.println(iResult+" is the frequency of number 11 in the given array.");

        aobj = null;
        sobj = null;

        System.gc();
    }

}
